src/client.ts

```diff
@@ -117,54 +120,47 @@ export class SymplifySDK {
             return null;
         }
 
+        const visitorID = ensureVisitorID(siteData, this.idGenerator);
```

Comment:
moved this up top to have less "threading" in the cases below (preview/allocation from cookie/new allocation)